test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -341,7 +489,7 @@ public void SchemaConnectionShouldntEnlist()
         }
 
         void AssertNoPreparedTransactions()
-            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0));
+            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0), "Prepared transactions found");
 
         int GetNumberOfPreparedTransactions()
```

Comment:
OK. I now understand better how durable vs. volatile could induce this kind of behavior change (and possibly create race conditions)... The MSDTC may introduce some asynchronicity which doesn't exist when using only System.Transactions (i.e. volatile-only). The question then would be how other provider implementations deal with this kind of thing, if at all.